client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/RemoteCacheSupport.java

```diff
@@ -59,6 +59,11 @@ public V putIfAbsent(K key, V value, long lifespan, TimeUnit unit) {
       return putIfAbsent(key, value, lifespan, unit, defaultMaxIdleTime, MILLISECONDS);
    }
 
+   @Override
+   public V putIfAbsent(K key, V value, long lifespan, TimeUnit lifespanUnit, long maxIdleTime, TimeUnit maxIdleTimeUnit) {
+      throw new UnsupportedOperationException();
```

Comment:
This seems wrong and unnecessary, since RemoteCacheImpl actually has a real implementation of this

Reply:
It is to add consistency to all of the default methods. It is implemented by the extending class so it is not used. But this means if this class is used by anything else it will have a proper default instead of the one in the JRE which is even worse imo.